tutorials/provisional/mosaic.py

```diff
@@ -0,0 +1,295 @@
+"""
+=======================================
+Complex and semantic figure composition
+=======================================
+
+.. warning::
+
+   This tutorial documents experimental / provisional API.
+   We are releasing this in v3.3 to get user feedback.  We may
+   make breaking changes in future versions with no warning.
+
+
+Laying out Axes in a Figure in a non uniform grid can be both tedious
+and verbose.  For dense even grids we have `.Figure.subplots` but for
+more complex layouts, such as Axes that span multiple columns / rows
+of the layout or leave some areas of the Figure blank you can use
+`.gridspec.GridSpec` (see :doc:`/tutorials/intermediate/gridspec`) or
+manually place your axes.  `.Figure.subplot_mosaic` aims to provide an
+interface to visually layout your axes (as either ASCII art or nested
+lists) to streamline this process.
+
+As a side effect of this interface, it is very natural to name your
+axes!  `.Figure.subplot_mosaic` returns a dictionary keyed on the
+labels used to layout the Figure.  By returning data structures with
+names it is easier to write plotting code that is independent of the
+Figure layout.
+
+
+This is inspired by a `proposed MEP
+<https://github.com/matplotlib/matplotlib/pull/4384>`__ and the
+`patchwork <https://github.com/thomasp85/patchwork>`__ library for R.
+While we do not implement the operator overloading style, we do
+provide a Pythonic API for specifying (nested) Axes layouts.
+
+"""
+import matplotlib.pyplot as plt
+import numpy as np
+
+
```

Comment:
I'd add
~~~
# Helper function used for visualization in the following examples
~~~

The function comes a bit out of nowhere and when reading top-to-bottom I was not clear why a function is defined here and if I should look at it / copy it to my code etc.

Reply:
Done.